release/nightly_tests/dataset/benchmark.py

```diff
@@ -50,32 +43,82 @@ def __init__(self, name):
         self.result = {}
         print(f"Running benchmark: {name}")
 
-    def run(self, name: str, fn: Callable[..., Dataset], **fn_run_args):
+    def run_materialize_ds(
+        self,
+        name: str,
+        fn: Callable[..., Dataset],
+        *fn_args,
+        **fn_run_args,
+    ):
+        """Run a benchmark on materializing a Ray Dataset. ``fn`` is expected to
+        return the Dataset which is to be materialized. Runtime and throughput
+        are automatically calculated and reported."""
+
         gc.collect()
 
         print(f"Running case: {name}")
         start_time = time.perf_counter()
-        output_ds = fn(**fn_run_args)
+        output_ds = fn(*fn_args, **fn_run_args)
         output_ds.materialize()
         duration = time.perf_counter() - start_time
 
         # TODO(chengsu): Record more metrics based on dataset stats.
-        self.result[name] = {BenchmarkMetric.RUNTIME.value: duration}
+        self.result[name] = {
+            BenchmarkMetric.RUNTIME.value: duration,
+            BenchmarkMetric.THROUGHPUT.value: output_ds.count() / duration,
+        }
+        print(f"Result of case {name}: {self.result[name]}")
+
+    def run_iterate_ds(
+        self,
+        name: str,
+        ds_iterator: Iterator[Any],
+        batch_size: int,
```

Comment:
How does this work if we want to do `batch_size=None`?

Reply:
removed the parameter, now we automatically calculate the size of each batch depending on the underlying type.